### ISSUE TITLE
Bug: relative url for links and images in package readme are not working

### ISSUE THREAD

**Issue by muescha** [gatsbyjs/gatsby#25999]
## Description

Relative URL for links and images, definition in package readme results in 404

### Steps to reproduce

Source of Readme: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-sharp/README.md#L167
Location: Just above https://www.gatsbyjs.org/packages/gatsby-plugin-sharp/#fit

### Screenshot

<img width="302" alt="Bildschirmfoto 2020-07-24 um 13 39 19" src="https://user-images.githubusercontent.com/184316/88387614-1eee5f80-cdb3-11ea-9516-2d8f0a8d48da.png">

### Examples

bad ones - point to files in the repo:
- `![Sharp transform fit options](./sharp-transform-fit-options.png)`
- `[index.js](./src/index.js)`

good ones:
- external full links -> `[overlayWith](http://sharp.pixelplumbing.com/en/stable/api-composite/#overlaywith)`
- points to gatsbyjs.org -> `[gatsby-image](/packages/gatsby-image/)`

### Expected result

- image should be displayed
- links should working

### Actual result

- images are not displayed
- link to source code not working

### Environment

gatsbyjs.org website

### Credits

reported by @Kaknut in:
- #25919 `Look's Like the image is missing`

**Comment by muescha** [comment]
### Analytics

the way for a package readme into the gatsby site is via the npm search in [gatsby-source-npm-package-search](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-source-npm-package-search). the readme is fetched from unpkg if there is no readme included.

so there is no defined `basepath` for the relative url in the readme

**Comment by kaknut** [comment]
Thanks for opening the issue and explaining it well.

**Comment by muescha** [comment]
### possible Solutions

### 1) force absolute path
- disallow relative paths 
- only allow full qualified paths

**pro:**
-> can be forced with a remark-lint rule on internal packages
-> no code change needed (except the markdown)

**con:**
-> only possible for "internal" packages 
-> hard to force on external packages -> since we can only change it with a PR


### 2) calculate source dir

- get the repo from package info 
- get from this repo the URL 
- with the relative path calculate a full path

**pro:**
-> can be used with every package readme

**con:**
-> more complicated to calculate
-> only possible for "internal" packages - since we can only do PR to other packages
-> not all packages have repo url, and links should be "changed" from repo to the CDN url for raw content (for github)

**Comment by dillionmegida** [comment]
Hello team, I'd like to mention a bug I faced similar to this:

Here's a package I created - [gatsby-remark-liquid-tags](https://www.gatsbyjs.org/packages/gatsby-remark-liquid-tags/)

The SERVICES.md file referenced is treated as a relative link to gatsby, hence a 404 page.

But [npm](https://www.npmjs.com/package/gatsby-remark-liquid-tags) treats it as an external link using the repo as the base URL. This is what I expect for gatsby too.

I'd love to hear you thoughts on this.
